dance/pipeline.py

```diff
@@ -992,20 +1026,39 @@ def get_step3_yaml(conf_save_path="examples/tuning/cta_svm/config_yamls/params/"
     conf_load_path = os.path.join(root_path, conf_load_path)
     result_load_path = os.path.join(root_path, result_load_path)
     conf = OmegaConf.load(conf_load_path)
-    result = pd.read_csv(result_load_path).sort_values(by=metric, ascending=ascending).head(top_k)
+    pipeline_top_k = default(step2_pipeline_planer.config.pipeline_top_k, DEFAULT_PIPELINE_TOP_K)
+    result = pd.read_csv(result_load_path).sort_values(by=metric, ascending=ascending).head(pipeline_top_k)
     columns = sorted([col for col in result.columns if col.startswith("pipeline")])
     pipeline_names = result.loc[:, columns].values
     count = 0
     for row in pipeline_names:
         pipeline = []
         row = [i for i in row]
-        for i, f in zip(required_indexes, required_funs):
-            row.insert(i, f)
         for x in row:
             for k in conf.pipeline:
                 if k["target"] == x:
                     pipeline.append(k)
+        for i, f in zip(required_indexes, required_funs):
```

Comment:
Some necessary functions, such as setConfig, can be obtained from the yaml of step 2 instead of the default yaml of step 3.